ci-operator/step-registry/cucushift/upgrade/eus/unpause-worker-mcp/cucushift-upgrade-eus-unpause-worker-mcp-commands.sh

```diff
@@ -19,32 +19,75 @@ function debug() {
     fi
 }
 
+# After upgrading, the paused mcp should not be upgraded, so its version is not same with normal mcp
+function pre_check_mcp(){
+    echo "Pre check, after upgrading, the paused mcp should not be upgraded..."
+    local paused_worker_version running_worker_version
+    paused_worker_version=$(oc get -l 'node-role.kubernetes.io/'${PAUSED_MCP_NAME}'=' nodes -ojson | jq '.items[0].status.nodeInfo' | jq -r '.containerRuntimeVersion,.kernelVersion,.osImage')
+    echo "Paused worker version: "
+    echo $paused_worker_version
+    running_worker_version=$(oc get -l 'node-role.kubernetes.io/master!=,node-role.kubernetes.io/'${PAUSED_MCP_NAME}'!=' nodes -ojson | jq '.items[0].status.nodeInfo' | jq -r '.containerRuntimeVersion,.kernelVersion,.osImage')
```

Comment:
if all worker nodes are labeled in paused pool, the ${running_worker_version} will be null,  then the check will pass even the paused worker node has been upgraded unexpectedly, right? so suggest to use running_master_version here.

Reply:
I guess there is a condition to control the behavior in https://github.com/openshift/release/pull/55328/files#diff-95606d20e2f14e072fb593b3bb53b8a0c0b4f1742526510152191446c15b315fR105

Reply:
`[[ "${PAUSED_MCP_NAME}" != "worker" ]]` only filter out eus upgrade with default paused pool `worker`, right? For canary upgrade, i guess there might be following scenarios:
 
```
 MCO_CONF_DAY2_CUSTOM_MCP: '[{"mcp_name": "mcpfoo", "mcp_node_num": "3", "mcp_node_label":
        "node-role.kubernetes.io/worker="}]'
      PAUSED_MCP_NAME: mcpfoo
```

```
 MCO_CONF_DAY2_CUSTOM_MCP: '[{"mcp_name": "mcpfoo", "mcp_node_num": "1", "mcp_node_label":
        "node-role.kubernetes.io/worker="}, {"mcp_name": "mcpbar", "mcp_node_num": "2", "mcp_node_label":
        "node-role.kubernetes.io/worker="}]'
      PAUSED_MCP_NAME: mcpfoo mcpbar
```

Reply:
`[[ "${PAUSED_MCP_NAME}" != "worker" ]]` I guess the original purpose for the condition is to ensure there are some multiple mcp in the cluster. If we container multi mcps and no-default-worker mcp cases into consideration,  I think we need to update the condition.

How about this:
1. filter out all the matched worker nodes by `PAUSED_MCP_NAME` setting.
2. If no more left worker nodes, then skip `pre_check_mcp` and `post_check_mcp` checking.

Reply:
hmm, based on jianl's [comment](https://github.com/openshift/release/pull/55328/files#r1722784213), node version's check should be a suggested step for canary upgrade, then i think we should keep it the same in all canary upgrade scenarios(including multi mcps and no-default-workers mcp) for better understanding and maintenance. 
what's more, do u think we could extend the check to eus(default-workers mcp) upgrade?

Reply:
I just implemented the multi mcp in new commit, let me run the tests